azkaban-common/src/main/java/azkaban/event/Event.java

```diff
@@ -53,24 +53,13 @@ public long getTime() {
     return time;
   }
 
-  public Object getData() {
+  public EventData getData() {
     return eventData;
   }
 
-  public static Event create(Object runner, Type type) {
-    return new Event(runner, type, null, true);
-  }
-
-  public static Event create(Object runner, Type type, Object eventData) {
-    return new Event(runner, type, eventData, true);
+  public static Event create(Object runner, Type type, EventData eventData) {
+    Preconditions.checkNotNull(eventData, "EventData was null");
```

Comment:
Added javadocs & stuff to `Event.create`. I don't see how compiler could enforce it, but I put there `@NotNull` annotation, which might help some tools / IDEs detect possible violations of that constraint.